lib/vanagon/platform.rb

```diff
@@ -456,7 +456,7 @@ def generate_compiled_archive(project)
         "mkdir #{archive_directory}",
         "gunzip -c #{name_and_version}.tar.gz | '#{tar}' -C #{archive_directory} -xf -",
         "rm #{name_and_version}.tar.gz",
-        "#{tar} cf #{name_and_version_and_platform}.tar -C #{archive_directory}/#{name_and_version} `#{find} #{archive_directory}/#{name_and_version} -maxdepth 1 -mindepth 1 -type d | sed -e 's|#{archive_directory}/#{name_and_version}/||'`",
+        "cd #{archive_directory}/#{name_and_version}; #{tar} cf ../../#{name_and_version_and_platform}.tar `#{find} . -type d`",
```

Comment:
Can you expand the tar flags? I'm guessing `cf` is `--copy --file`? but I don't know for sure. I'm adopting @e-gris's preference for explicitness. 😊

Reply:
No, we can't, we need to use the short flags for compatibility. `c` is `--create` and `f` is `--file` in gnu tar but there are no long options for bsd tar

Reply:
oooh

Reply:
I have a question of the find command changes. The first you're only grabbing the top level set of directories, and the second, you're grabbing all of the directories. When I run this locally, I get very different results. Is this intentional?

```
[0] melissa@mrt:~$ find ci-job-configs -maxdepth 1 -mindepth 1 -type d
ci-job-configs/.git
ci-job-configs/.github
ci-job-configs/doc
ci-job-configs/jenkii
ci-job-configs/puppet
ci-job-configs/register-repo
ci-job-configs/resources
ci-job-configs/utils
[0] melissa@mrt:~$ cd ci-job-configs ; find . -type d
.
./.git
./.git/refs
./.git/refs/heads
./.git/refs/tags
./.git/refs/remotes
./.git/refs/remotes/origin
./.git/branches
./.git/hooks
./.git/info
./.git/objects
./.git/objects/pack
./.git/objects/info
./.git/logs
./.git/logs/refs
./.git/logs/refs/remotes
./.git/logs/refs/remotes/origin
./.git/logs/refs/heads
./.github
```

Reply:
The directory prefix and getting rid of sed I'm totally on board with. I'm worried about the content disparities. Also, the last ci-job-configs example wasn't a full grab, I just pasted the top few lines

Reply:
Reinforcing what @underscorgan said: explicit is nice but portability can put the kibosh on it. If the script is expected to work on non-GNU systems (Mac and BSD, for instance), care must be taken to stick with common forms.

Reply:
I had portability issues with `mindepth` and `maxdepth`. `tar` will add things recursively, so I thought that should end up with the same output, though with a quick test I ended up with duplicates when I was adding all the directories, might try to '*'

```
morgan@river.corp.puppetlabs.net % tar cf test2.tar foo foo/bar foo/bar/baz
10:29:19: /tmp
morgan@river.corp.puppetlabs.net % tar tf test2.tar
foo/
foo/bar/
foo/bar/baz/
foo/bar/
foo/bar/baz/
foo/bar/baz/
```